docs/madr/decisions/055-meshservice-subset.md

```diff
@@ -0,0 +1,175 @@
+# Subsets of Dataplanes per MeshService
+
+* Status: accepted
+
+## Context and Problem Statement
+
+Referring to a subset of endpoints for a given Service is important for service
+mesh. It's most commonly use to implement things like blue/green deployments or canary
+deployments.
+
+For example if we have a `backend` service that we want to upgrade from `v1` to
+`v2`. With `kuma.io/service`, this would mean having `Dataplanes` with
+`kuma.io/service: backend` and an additional tag either `version: v1` or
+`version: v2`. In order to do weighted routing to these different versions, a
+user would create a `MeshHTTPRoute`:
+
+```yaml
+        backendRefs:
+        - kind: MeshServiceSubset
+          tags:
+            version: v2
+          name: backend
+          weight: 10
+        - kind: MeshServiceSubset
+          tags:
+            version: v1
+          name: backend
+          weight: 90
+```
+
+Now that we have `MeshService` objects, the user has something like the following:
+
+```yaml
+kind: MeshService
+spec:
+  selector:
+    dataplaneTags:
+      app: backend
+```
+
+and matched `Dataplanes` have `version` tags set.
+
+The question this MADR answers is how do users handle the same concept with
+`MeshService`?
+
+## Considered Options
+
+* Support `MeshServiceSubset` with `MeshService` objects
+* Users create additional `MeshServices`
+
+## Decision Outcome
+
+Chosen option: additional `MeshServices`, no more `MeshServiceSubset`
```

Comment:
On Kubernetes, we automatically generate a `MeshService` based on a `Service` object. Can a user achieve the same result by creating a `Service` that selects apps with version v1?

Reply:
yeah that's the foundation of making it cohesive with K8s, I'll call it out explicitly though

Reply:
is that enough? the rest of the document just sort of follows if you remember that the Service -> MeshService conversion happens.